File: RemoveDuplicatesFromList/recursive.java
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head == null || head.next == null) {
            return head;
        }
        ListNode distinct = head;
        while(distinct.next != null && distinct.next.val == head.val) {
            distinct = distinct.next;
        }

        head.next = deleteDuplicates(distinct.next);

        return head;
    }
}

File: RemoveDuplicatesFromList/recursive.py
class Solution:
    # @param head, a ListNode
    # @return a ListNode
    def deleteDuplicates(self, head):
        if head and head.next and head.val != head.next.val:
            head.next = self.deleteDuplicates(head.next)
        elif head and head.next:
            head = self.deleteDuplicates(head.next)
        return head
